public/javascript/controllers/application.js

```diff
@@ -56,61 +33,47 @@ function bindClicks() {
 }
 
 function updateViews() {
-    updateDesktopVisbility();
-    viewer.updateViewer(activeMode, modes[activeMode]);
-    buttonUtil.markSelected('switchmode', activeMode);
 
-    if (activeMode === 'reader') {
-        buttonUtil.removeClassFromAttributeNameAndValue('showreaderactive', 'true', 'is-hidden');
-    } else {
-        buttonUtil.addClassToAttributeNameAndValue('showreaderactive', 'true', 'is-hidden');
-    }
-}
+    document.body.className = 'is-' + activeMode;
 
-function updateMode(newMode) {
-    var oldMode = activeMode;
-
-    if (newMode === 'desktop' && !desktopEnabled && window._previewEnv !== 'live') {
-        return;
+    if (desktopEnabled) {
+        document.body.className += ' desktop-enabled';
     }
 
+    viewer.updateViewer(activeMode, modes[activeMode]);
+    buttonUtil.markSelected('switchmode', activeMode);
+}
+
+function triggerAnalytics(oldMode, newMode) {
     if ((oldMode !== newMode) && modes[oldMode].isMobile && modes[newMode].isMobile) {
         analyticsCtrl.recordOrientationChange();
     }
 
-    if ((oldMode !== "desktop") && (newMode === "desktop")) {
+    if ((oldMode !== 'desktop') && (newMode === 'desktop')) {
         analyticsCtrl.recordDesktopViewed();
     }
 
-    if ((oldMode !== "reader") && (newMode === "reader")) {
+    if ((oldMode !== 'reader') && (newMode === 'reader')) {
         analyticsCtrl.recordReaderMode();
     }
 
     if ((oldMode !== newMode) && !modes[oldMode].isMobile && modes[newMode].isMobile) {
         analyticsCtrl.recordMobileViewed();
     }
-
-    activeMode = newMode;
-
-
-    updateViews();
 }
 
-function updateDesktopVisbility() {
-    if (desktopEnabled) {
-        buttonUtil.addClassToAttributeNameAndValue('toggledesktop', 'true', 'is-checked');
-        buttonUtil.removeClassFromAttributeNameAndValue('switchmode', 'desktop', 'is-hidden');
-        buttonUtil.addClassToAttributeNameAndValue('hidedesktopenabled', 'true', 'is-hidden');
+function updateMode(newMode) {
+    var oldMode = activeMode;
 
-    } else {
-        buttonUtil.removeClassFromAttributeNameAndValue('toggledesktop', 'true', 'is-checked');
-        buttonUtil.addClassToAttributeNameAndValue('switchmode', 'desktop', 'is-hidden', 'none');
-        buttonUtil.removeClassFromAttributeNameAndValue('hidedesktopenabled', 'true', 'is-hidden');
+    if (newMode === 'desktop' && !desktopEnabled && window._previewEnv !== 'live') {
```

Comment:
Well It needs to be passed from the server side to the client side code. But I'll move it out of here so it's extracted on init (along with any other config variables that come along)